src/mud/examples/poisson.py

```diff
@@ -196,95 +282,42 @@ def run_2d_poisson_sol(
-
-        # Label and format figure
-        _ = plt.xlim(0, 1)
-        _ = plt.ylim(0, 1)
-        _ = plt.title("Response Surface")
-        _ = plt.xlabel("$x_1$")
-        _ = plt.ylabel("$x_2$")
-
-        ax = fig.add_subplot(1, 2, 2)
-        # Plot closest solution in sample set to the reference solution
-        plot_solution_spline(
-            closest,
-            ax=ax,
-            lw=5,
-            c="green",
-            ls="--",
-            label=r"$\hat{g}(\lambda^\dagger)$",
-            zorder=50,
-        )
-        # Plot first 100 lambda initial
-        for i, lam in enumerate(poisson_prob.lam[0:50]):
-            plot_solution_spline(
-                lam,
-                plot_true=False,
-                ax=ax,
-                lw=1,
-                c="purple",
-                alpha=0.1,
-            )
-
-        plot_vert_line(ax, 1 / 3.0, color="b", linestyle="--", alpha=0.6)
-        plot_vert_line(ax, 2 / 3.0, color="b", linestyle="--", alpha=0.6)
-
-        ax.set_title("Boundary Condition")
-        ax.set_ylabel("")
-        ax.legend(
-            ["$g(x_2)$", r"$\hat{g}(x_2,\lambda^\dagger)$", r"$\hat{g}(x_2,\lambda_i)$"]
-        )
-
-        fig.tight_layout()
-
-        save_figure(
-            "response_surface",
-            save_path,
+        ax = plot_response_ex(
+            poisson_prob,
+            closest=closest,
+            raw_data=raw_data,
+            group_idxs=group_idxs,
+            markers=markers,
+            sensors_mask=idx_o,
+            colors=colors,
+            save_path=save_path,
             close_fig=close_fig,
             dpi=dpi,
             bbox_inches="tight",
         )
         axes.append(ax)
     if "qoi" in plot_fig or "all" in plot_fig:
-        fig = plt.figure(figsize=(10, 5))
-        for i in range(num_components):
-            ax = fig.add_subplot(1, 2, i + 1)
-            mud_prob.plot_params_2d(ax=ax, y=i, contours=True, colorbar=True)
-            if i == 1:
-                ax.set_ylabel("")
-        save_figure(
-            "learned_qoi", save_path, close_fig=close_fig, dpi=dpi, bbox_inches="tight"
+        ax = plot_qoi_ex(
+            mud_prob,
+            num_components=num_components,
+            save_path=save_path,
+            close_fig=close_fig,
+            dpi=dpi,
+            bbox_inches="tight",
         )
         axes.append(ax)
     if "densities" in plot_fig or "all" in plot_fig:
         ax1 = mud_prob.plot_param_space(param_idx=0, **param1_kwargs)
-        save_figure(
-            "lam1", save_path, close_fig=close_fig, dpi=dpi, bbox_inches="tight"
-        )
+        if save_path:
+            save_figure(
+                "lam1", save_path, close_fig=close_fig, dpi=dpi, bbox_inches="tight"
+            )
 
         ax2 = mud_prob.plot_param_space(param_idx=1, **param2_kwargs)
-        save_figure(
-            "lam2", save_path, close_fig=close_fig, dpi=dpi, bbox_inches="tight"
-        )
+        if save_path:
```

Comment:
Should be 
`if save_path is not None:`
See [PEP 8](https://peps.python.org/pep-0008/)